src/physics.h

```diff
@@ -27,7 +27,7 @@
 static const float oocc = (float) 0.0000000000000000111265005605; //1/c^2
 static const float c    = (float) 299792458.0;
 static const float co10 = (float) 29979245.8;
-float copysign( float x, float y );
+// float copysign( float x, float y );
```

Comment:
I'm guessing you did this because it conflicts with the stdlib.

In those cases commenting the code isn't a good solution. A simple `#ifndef` may work, and/or a configure test to check whether it's available would be the proper approach.

Or just deleting it. AFAIK this was introduced in C99, which should be widely supported by now, so we may no need to keep compatibility with older compilers that lack C99 support. In any case you shouldn't leave commented code everywher (I know we did this already, but it's wrong).

I'm not sure about windows (MSVC) support, but I'm guessing it should be there.

Reply:
For MSVC compat, I would recommend making a header file that can be used to explicitly provide compatibility; sometimes functions just need to be renamed; sometimes they need wrappers to ensure the same interface. Fortunately, MS has gotten better about compatibility with the standards - at least providing the functions; but they've gotten worse about implementation details (f.e snprintf vs _snprintf - there's differences in behavior that need to be accounted for, and MS favors s_snprintf which is also different); not sure how much you'll run into that with copysign, hopefully not at all.

$0.02 FWIW.

Reply:
Yes, I did this because it was already declared, evidently in stdlib. If this was introduced in C99, I think it would be fine to take out these declarations altogether. (That standard is now 20 years old. ;-) )

Reply:
Removed altogether.

Reply:
MSVC is kind of its own standard... depending on what version you're using. MS has gotten better about adhering to ISO standard C/C++ (starting with VS2008 which finally introduced `stdint.h` and slowly getting better each iteration since) but it can still be hit-or-miss on some things last I knew...but I haven't used VS in quite a while now so may be it's no longer an issue...

Reply:
> The Universal C Runtime library (UCRT) provides many integral and floating-point math library functions, including all of those required by ISO C99.

[https://docs.microsoft.com/en-us/cpp/c-runtime-library/floating-point-support?view=vs-2019](https://docs.microsoft.com/en-us/cpp/c-runtime-library/floating-point-support?view=vs-2019)

Reply:
So it sounds like Visual Studio 2019, at least, supports `copysign` as required by the standard.

Reply:
Unfortunately they dropped docs for VS2013 and older and stop listing what version introduced something, so it's hard to tell where support began. My guess is they're only supporting VS2015 and newer (right now); that said, many folks may still have access to older compilers...one can only hope folks use a newer compiler. Certainly would be good to say "using this version and newer" if it doesn't already.